documentation/src/test/java/example/exception/AssertDoesNotThrowExceptionDemoTestCase.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example.exception;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import org.junit.jupiter.api.Test;
+
+// @formatter:off
+// tag::user_guide[]
+
+class AssertDoesNotThrowExceptionDemoTestCase {
```

Comment:
I realize this is not officially documented, but test classes in the `documentation` module must be named `*Demo`.

https://github.com/junit-team/junit5/blob/3e03c97f83d1ff35dcb2b3a7cc48b54943eb1b69/documentation/documentation.gradle.kts#L160

So, please rename all `*TestCase` example classes to `*Demo` so that they are run with the Gradle build.

Reply:
Thanks for the suggestion. I initially named the classes as `*Demo` but that was failing the build because of `UncaughtExceptionHandlingDemoTestCase` (which was designed to fail). So I renamed it and others (to keep parity) to `*TestCase` to exclude from build. If I rename then to `*Demo` then the build will fail. 

One thing I can do is I can rename all the tests except `UncaughtExceptionHandlingDemoTestCase` to make the builds pass. do you think that's a good idea? do you have any other suggestions?

Reply:
> do you have any other suggestions?

Yes. 😉 

Please name them all `*Demo` and make use of our `@ExpectToFail` annotation/extension.

See `example.AssertionsDemo.timeoutExceeded()` for details.

Reply:
https://github.com/junit-team/junit5/blob/3e03c97f83d1ff35dcb2b3a7cc48b54943eb1b69/documentation/src/test/java/example/AssertionsDemo.java#L131-L143

Reply:
ah this is nice and exactly what I wanted. Thanks for the suggestion. I have changed all the classes name to `*Demo`